repository/fetch-repos.go

```diff
@@ -91,19 +91,17 @@ func getReposByOrg(config *config.GitXargsConfig) ([]*github.Repository, error)
 		}
 
 		// github.RepositoryListByOrgOptions doesn't seem to be able to filter out archived repos
-		// So re-slice the repos list if --skip-archived-repos is passed and the repository is in archived/read-only state
-		for i, repo := range repos {
+		// So filter the repos list if --skip-archived-repos is passed and the repository is in archived/read-only state
+		for _, repo := range repos {
```

Comment:
```
if config.SkipArchivedRepos {
	for _, repo := range repos {
		if repo.GetArchived() {
			logger.WithFields(logrus.Fields{
				"Name": repo.GetFullName(),
			}).Debug("Skipping archived repository")

			// Track repos to skip because of archived status for our final run report
			config.Stats.TrackSingle(stats.ReposArchivedSkipped, repo)
		} else {
			reposToAdd = append(reposToAdd, repo)
		}
	}
} else {
	reposToAdd = repos
}	
```
how about checking for the flag first, cut down on the extra reassignments if `config.SkipArchivedRepos == false`?